library/network_connections.py

```diff
@@ -1061,6 +1061,56 @@ def volatilize_connection(self, connection, timeout=10):
                 % (cb_args.get("error", "unknown error"), cb_args)
             )
 
+    def create_checkpoint(self):
+        """ Create a new checkpoint """
+        checkpoint = self._call_checkpoint(
+            "create",
+            [],  # devices, empty list is all devices
+            60,  # timeout in seconds, FIXME: Make it an option
+            Util.NM().CheckpointCreateFlags.DELETE_NEW_CONNECTIONS
+            | Util.NM().CheckpointCreateFlags.DISCONNECT_NEW_DEVICES,
+        )
+
+        if checkpoint:
+            return checkpoint.get_path()
+
+    def destroy_checkpoint(self, path):
+        """ Destroy the specified checkpoint """
+        self._call_checkpoint("destroy", path)
+
+    def rollback_checkpoint(self, path):
+        """ Rollback the specified checkpoint """
+        self._call_checkpoint("rollback", path)
+
+    def _call_checkpoint(self, action, *args):
+        action_name = "checkpoint_" + action
+        finish = action_name + "_finish"
+        callback = Util.create_callback(finish)
+        cancellable = Util.create_cancellable()
+        user_data = {}
+
+        mainloop_timeout = 10
+
+        action_method = getattr(self.nmclient, action_name)
+        fullargs = []
+        fullargs += args
+        fullargs += (cancellable, callback, user_data)
+
+        action_method(*fullargs)
+
+        if not Util.GMainLoop_run(mainloop_timeout):
```

Comment:
is having these local variables really preferable? If you don't like `if not Util.GMainLoop_ru(10):`, then `if not Util.GMainLoop_run(timeout = 10):` seems still better.

The same could be said for other local variables. Too much verbosity does not make code more legible, on the contrary it just clutters.

Reply:
Usually I like the variables more but in this case some were not needed, changed it.